# HW2

In [1]:
name = "Leonid"              # Enter your name
user = "rempell"              # Enter your BC username
print("HW2 submission for {} ({})".format(name,user))

HW2 submission for Leonid (rempell)


Click on HW2 at the top of the page, immediately to the right of Jupyter. This will open a window that contains the line "Enter a new notebook name:". Please rename the file so that it includes your BC username. 

For example, I would change the file name from <b>HW2</b> to <b>HW2-reuterj</b>.

### 1. Variable Types and Print Statements

In [4]:
w = 'Douglas Adams'
x = "123456"
y = 42.
z = 3

<b>(1 point) 1a.</b> Write code that prints the variable type for 'x', 'y', and 'z'. For example, the first row of output should be "x is of type <class 'str'>".

In [4]:
print(type(x)) #He wanted in format of "x is of type <class 'str'>'"
print(type(y)) 
print(type(z))

<class 'str'>
<class 'float'>
<class 'int'>


<b>(1 point) 1b.</b> Write code that prints the 'x' three times in a row using variables 'x' and 'z' and the "*" operator.

In [5]:
print(x*z)

123456123456123456


<b>(1 point) 1c.</b> Create new variable 'y_str' which converts 'y' into a string and new variable 'z_str' which converts 'z' into a string and then write a print statement that prints the length of each string on the same line.

In [6]:
y_str = str(y) 
z_str = str(z)
print(len(y_str), len(z_str))

4 1


<b>(1 point) 1d.</b>  Write code that performs the following calculation 

new_num = $(123456/42)^3$

using versions of 'x', 'y', and 'z' and then prints "'new_num' equals " (without the ""s) followed by the answer.

In [9]:
new_num = (float(x)/y)**3
print('new_num equal ' + str(new_num))

new_num equal 25397369279.813416


<b>(1 point) 1e.</b> Write a print statement using variables 'w' and 'y' to generate the following output, including the single quotes:

Douglas Adams mixed 42(!) 'Pan Galactic Gargle Blasters' for the Interplanetary World Congress

Hint: You will want to use print("<i>your text here including two {}s</i>".format(w,y))</i>. You will also want to make sure that your code prints 42 instead of 42.0.

In [14]:
print("{} mixed {}(!) 'Pan Galactic Gargle Blasters' for the Interplanetary World Congress".format(w,int(y)))

Douglas Adams mixed 42(!) 'Pan Galactic Gargle Blasters' for the Interplanetary World Congress


### 2. Slicing Strings

In [15]:
a = 'Roy G Biv'

<b>Example.</b> How could you slice 'a' so that you are left with the two letters: 'Ro'?

In [16]:
a[0:2]         # this slice begins with element 0 and ends before element 2

'Ro'

<b>(1 point) 2a.</b> How could you slice 'a' so that you are left with the last five characters: 'G Biv'?

In [21]:
a[4:]

'G Biv'

<b>(1 points) 2b.</b> How could you slice 'a' so that you are left with uppercase versions of the three letters: 'Roy' and lowercase versions of the three letters "Biv"? (Hint: You can slice twice.)

In [26]:
a.upper()[0:3] + a.lower()[6:] #You could also slice first and then uppercase or lowercase which is more effcicent
                               #Usually you want to apply the filter first before you mess with the data, so uppercase first is more proper

'ROYbiv'

<b>(1 points) 2c.</b> In the header to question 1, I defined variable 'x' to be a string variable equal to "123456" and 'z' to be the integer 3. Write code that uses 'x' and 'z' to creates new_str = "123456123456123456". Then, create 'new_int' by slicing <b>elements</b> 4 through 10 of 'new_str' and convert into an integer. Print 'new_int'.

Hint: In Python, the very first element of a string is indexed as element zero.

In [28]:
new_str = str(x*z)
new_int = int(new_str[4:11]) #when you say elements 4 through 10 do you mean including elements 4 and 10? this is the code for that, if 10 is not included then it is just 4:10
new_int

5612345

In [9]:
print(int((x*z)[4:11])) #written in one line but harder to debug and grade

5612345


### 3. Your First For Loops and If Statements

Below is a string containing letters and numbers, which will serve as the input to the loop.

In [10]:
loop_str = "a1B2c2D4e5f1g0h4I3"

<b>(1 points) 3a.</b> Write code that uses print statements to print the uppercase version of each letter on its own line but that does not print any of the numbers.

Hint: What does the code char.isdigit() return? Can you use this to decide what to print?

In [44]:
for a in loop_str:
    if a.isdigit() != 1: #better to use .isalpha() because this code would return a space if it is there
        print(a.upper())

A
B
C
D
E
F
G
H
I


<b>(1 points) 3b.</b> Write code that adds up all of the numbers contained in 'loop_str' and prints the value of this new variable 'total' at the end of the loop.

In [50]:
# Your code goes in this cell
# I recommend using "total = 0" as the first line of code and "for char in loop_str:" as the second line of code
# The print statement should include "total = " and should not be included within the loop

total=0
for char in loop_str:
    if char.isdigit() == 1:
        total = total + 1 #prof wanted total = total + int(char) because he did not want a count, he wanted the actual total of the numbers in the string
print("total=" + str(total))

total=9


### Finally, click on the disk icon to save your notebook, and then choose "close and halt" from the File menu.

You can now upload a Jupyter Notebook (.ipynb) copy of your notebook to Canvas. In my case, I would upload "<b>HW2-reuterj.ipynb</b>".